helix-view/src/document.rs

```diff
@@ -185,6 +185,9 @@ pub struct Document {
 
     // when document was used for most-recent-used buffer picker
     pub focused_at: std::time::Instant,
+
+    // logically true when equal to Some(true); None implies we haven't checked yet
+    pub readonly: Option<bool>,
```

Comment:
I don't think `None` should exist here. We should always recheck during certain actions since the FSA could change at any point. So we should just init to false here and the always rechdck